tests/utils/test_remote_uploader.py

```diff
@@ -20,11 +20,14 @@ class DummyObjectStore(ObjectStore):
     """Dummy ObjectStore implementation that is backed by a local directory."""
 
     def __init__(self, **kwargs: Dict[str, Any]) -> None:
-        self.tmp_dir = tempfile.TemporaryDirectory()
+        self.tmp_dir = self.get_tmp_dir()
         self.root = self.tmp_dir.name
         self.sleep_sec = 0
         self.dest_filename = ''
 
+    def get_tmp_dir(self):
```

Comment:
Why is this necessary instead of just using it directly? Other files (e.g. tests/trainer/test_checkpoint.py) have it as an inline private function

Reply:
it's for mock and overwrite the temp path [here](https://github.com/mosaicml/composer/pull/3376/files#diff-17a1d776b94db0772f9f529b7506f222bc77c571df37cf4974099bac5ec0c128R799)